promdump/promdump.go

```diff
@@ -642,18 +763,81 @@ func main() {
 			// ['export_type="node_export"', 'node_prefix="yb-dev-..."'] => '{export_type="node_export",node_prefix="yb-dev-..."}'
 			ybMetric = fmt.Sprintf("{%v}", strings.Join(labels, ","))
 
-			err = exportMetric(ctx, promApi, ybMetric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, metricName)
+			fileCount, err = exportMetric(ctx, promApi, ybMetric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, metricName)
 
 			if err != nil {
 				log.Printf("exportMetric: export of metric %v failed with error %v; moving to next metric", metricName, err)
 				continue
 			}
+			v.fileCount = fileCount
+			fileCount = 0
 		}
 	}
 	if *metric != "" {
-		err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
+		customMatricCount, err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
 		if err != nil {
 			log.Fatalln("exportMetric:", err)
 		}
 	}
+
+	// tar the file when Tar feature is enable
+
+	if *enableTar {
+		// Set default filename if not provided
+
+		var filename string
+
+		if *tarFilename == "" {
+			// If filename is not provided, generate a default filename
+			filename = generateDefaultFilename()
+		} else {
+			filename = *tarFilename
+			// Given  filename is properly not given (missing extension)
+			tarExtension := ".tar.gz"
+			bz2Extension := ".bz2"
+
+			fileExtension := strings.HasSuffix(filename, tarExtension) || strings.HasSuffix(filename, bz2Extension)
+
+			if !fileExtension {
+
+				fmt.Println("Please prodive proper filename with extension \n")
+
+				return
+
+			}
+
+			// If filename is provided and it already exists, generate a default filename
+			if _, err := os.Stat(filename); err == nil {
+				defaultFilename := generateDefaultFilename()
+				if filename != defaultFilename {
+					fmt.Println(filename, "already exists using default filename \n")
+					filename = defaultFilename
+				}
+			}
+		}
+
+		out, err := os.Create(filename)
+		if err != nil {
+			log.Fatalln("Error writing archive:", err)
+		}
+		// Defer file closure immediately after creating the file with error handling
+		defer func() {
+			if cerr := out.Close(); cerr != nil {
+				log.Println("Error closing file:", cerr)
+			}
+		}()
+
+		tw := tar.NewWriter(out)
+		defer tw.Close()
+
+		// Create the archive and write the output to the "out" Writer
```

Comment:
Does this give any information not already available by reading the code? Might be better as a debug log entry.